### PR TITLE
MRG, DOC: Clarify lambda/SNR

### LINKED ISSUE #10149: Clarification on the relation between SNR and regularization factor
#### Proposed documentation enhancement
Could we reconcile the different mentions of the relation between SNR and regularization factor for inverse solution? Following are couple of places that don't agree with each other: 

- https://mne.tools/stable/overview/implementation.html#whitening-and-scaling says \lambda^2 ~ 1 / SNR
- https://github.com/mne-tools/mne-python/blob/c3ff05c05c7bfcf765249f27a517803f27f48b4e/mne/minimum_norm/inverse.py#L532 recommends \lambda^2 ~ 1 / SNR^2 
- https://mne.tools/stable/auto_examples/inverse/mixed_source_space_inverse.html?highlight=snr#compute-inverse-solution also uses \lambda^2 ~ 1 / SNR^2
- https://mne.tools/stable/generated/mne.minimum_norm.estimate_snr.html?highlight=snr#mne-minimum-norm-estimate-snr docstring says \lambda ~ 1 / SNR^2

It will be helpful too if there's a reference for this relation.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #10149

Make everything consistent by just talking about SNR of amplitudes, and correct an error in `estimate_snr`

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
thx @larsoner

File: doc/_includes/inverse.rst [hunk 1]
```diff
@@ -156,8 +156,8 @@ distribution.
 
 A convenient choice for the source-covariance matrix :math:`R` is such that
 :math:`\text{trace}(\tilde{G} R \tilde{G}^\top) / \text{trace}(I) = 1`. With this
-choice we can approximate :math:`\lambda^2 \sim 1/SNR`, where SNR is the
-(power) signal-to-noise ratio of the whitened data.
+choice we can approximate :math:`\lambda^2 \sim 1/rm{SNR}^2`, where SNR is the
```
Comment (drammock) [inline doc/_includes/inverse.rst:159]: I think there's a missing backslash before `rm` here

https://40144-1301584-gh.circle-artifacts.com/0/dev/overview/implementation.html?highlight=convenient%20choice#whitening-and-scaling

Comment (drammock) [reply doc/_includes/inverse.rst:159]: dang, I was 1 minute too slow

Comment (larsoner) [reply doc/_includes/inverse.rst:159]: Will push to `main`

**Pull request merged**